bin/yugabyted

```diff
@@ -1769,52 +1821,204 @@ class YBAdminProxy(object):
 
 # Proxy for ysqlsh commands.
 class YsqlProxy(object):
-    def __init__(self, port, path=os.path.join(BIN_DIR, YUGABYTE_API_CLIENT_PROGRAMS["ysql"])):
+    def __init__(self, ip, port, path=os.path.join(BIN_DIR, YUGABYTE_API_CLIENT_PROGRAMS["ysql"]),
+                    use_env_credentials=True):
+        self.ip = str(ip)
         self.port = str(port)
         self.path = path
+        self.username, self.password, self.db = get_env_credentials(
+                                                        YUGABYTE_API_CLIENT_PROGRAMS["ysql"],
+                                                        use_env_credentials)
 
     # Starts interactive YSQL shell.
     def connect(self, db=None):
-        cmd = [self.path, "-p", self.port]
-        if db:
-            cmd.extend(["-d", db])
-        shell = subprocess.Popen(cmd)
+        if db is None:
+            db = self.db
+        cmd = [self.path, "-h", self.ip, "-p", self.port, "-d", db]
+        shell = subprocess.Popen(cmd, env={"PGUSER": self.username})
         shell.communicate()
 
     # Checks if db exists.
     # Note that this will return false if ysqlsh can't connect, even if db exists.
     def db_exists(self, db):
-        cmd = [self.path, "-p", self.port, "-q", "-c", "\\t", "-c",
+        cmd = [self.path, "-h", self.ip, "-p", self.port, "-q", "-c", "\\t", "-c",
             "select datname from pg_catalog.pg_database where datname='{}'".format(db)]
-        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+            env={"PGUSER": self.username, "PGPASSWORD": self.password, "PGDATABASE": self.db})
         out, err = p.communicate()
         return out.strip() == db
 
     # Creates specified db.
     def create_db(self, db):
-        cmd = [self.path, "-p", self.port, "-c", "create database {}".format(db)]
-        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        cmd = [self.path, "-h", self.ip, "-p", self.port, "-c", "create database {}".format(db)]
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
```

Comment:
make a utility function that does the subprocess.Popen with the env vars so create_db looks simple like `run_ysql_command("create database. ..")`